.github/workflows/build-pr.yml

```diff
@@ -0,0 +1,95 @@
+---
+name: build-pr
+on:
+  push:
+    branches:
+      - "build-pr-*"
+      - "ios-pr-*"
+      - "android-pr-*"
+
+env:
+  NODE_VERSION: 18.7.0
+  TERM: xterm
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/test
+        uses: ./.github/actions/test
+
+  build-ios-pr:
+    runs-on: macos-12
+    if: ${{ !startsWith(github.ref_name, 'android') }}
+    needs:
+      - test
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
```

Comment:
Just for my knowledge... what are we checking out here?

Reply:
We are checking out the branch/sha that triggered this workflow